flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/cli/CliTableauResultView.java

```diff
@@ -276,13 +277,13 @@ private void printSingleRow(int[] colWidths, String[] cols) {
 		sb.append("|");
 		int idx = 0;
 		for (String col : cols) {
-			byte[] colBytes = getUTF8Bytes(col);
 			sb.append(" ");
-			if (colBytes.length <= colWidths[idx]) {
-				sb.append(StringUtils.repeat(' ', colWidths[idx] - colBytes.length));
+			int colWidth = getStringWidth(col);
+			if (colWidth <= colWidths[idx]) {
+				sb.append(StringUtils.repeat(' ', colWidths[idx] - colWidth));
 				sb.append(col);
 			} else {
-				sb.append(subMaxString(col, colWidths[idx]));
+				sb.append(getFixedString(col, colWidths[idx]));
```

Comment:
`getFixedString` => `truncateString`?

Reply:
and second parameter can be replaced with `colWidths[idx] - COLUMN_TRUNCATED_FLAG_WIDTH` as `targetLength`